components/core/src/clp_s/ArchiveWriter.hpp

```diff
@@ -95,8 +97,42 @@ class ArchiveWriter {
      * @param key
      * @return the node id
      */
-    int32_t add_node(int parent_node_id, NodeType type, std::string_view const key) {
-        return m_schema_tree.add_node(parent_node_id, type, key);
+    int32_t add_node(int parent_node_id, NodeType type, std::string_view key) {
+        auto const node_id{m_schema_tree.add_node(parent_node_id, type, key)};
+        if (m_matched_timestamp_prefix_node_id == parent_node_id
+            && m_authoritative_timestamp.size() > 0)
+        {
+            if (constants::cRootNodeId == parent_node_id) {
+                if (NodeType::Object == type) {
+                    m_matched_timestamp_prefix_node_id = node_id;
+                }
+            } else if ((m_authoritative_timestamp.size() - m_matched_timestamp_prefix_length) > 1) {
+                if (NodeType::Object == type
+                    && m_authoritative_timestamp.at(m_matched_timestamp_prefix_length) == key)
+                {
+                    m_matched_timestamp_prefix_length += 1;
+                    m_matched_timestamp_prefix_node_id = node_id;
+                }
+            }
+        }
```

Comment:
Not sure if we can write it in this way.
```suggestion
        if (NodeType::Object == type &&
            m_matched_timestamp_prefix_node_id == parent_node_id
            && m_authoritative_timestamp.size() > m_matched_timestamp_prefix_length + 1)
        {
            if (constants::cRootNodeId == parent_node_id) {
                    m_matched_timestamp_prefix_node_id = node_id;
            } else if (m_authoritative_timestamp.at(m_matched_timestamp_prefix_length) == key) {
                    m_matched_timestamp_prefix_length += 1;
                    m_matched_timestamp_prefix_node_id = node_id;
            }
```

Reply:
I guess per discussion in the other comment are you fine leaving this as is for now and considering refactoring in a follow-up PR?